app/logic/minor.py

```diff
@@ -68,34 +111,171 @@ def getProgramEngagementHistory(program_id, username, term_id):
     # engagement's respective type, name, id, and term.
     terms = defaultdict(list)
     for course in courses:
-        terms[(course.term.description, course.term.id)].append({"name":course.courseName, "id":course.id, "type":"course", "term":course.term})
+        terms[(course.term.description, course.term.id)].append({
+                    "name":course.courseName, 
+                    "id":course.id, 
+                    "type":"course", 
+                    "matched": course.reqMatch,
+                    "term":course.term.id})
+
+    programMatchCase = Case(None, [(IndividualRequirement.program.is_null(True) , 0)], 1)
 
-    events = (Event.select(Event, Program)
+    events = (Event.select(Event, Program, programMatchCase.alias('reqMatch'))
                    .join(EventParticipant, on=(Event.id == EventParticipant.event)).switch()
                    .join(Program)
+                   .join(IndividualRequirement, JOIN.LEFT_OUTER, on=((IndividualRequirement.program == Program.id) &
+                                                                     (IndividualRequirement.username == EventParticipant.user) &
+                                                                     (IndividualRequirement.term == Event.term)))
                    .where(EventParticipant.user == username)
                    .group_by(Event.program, Event.term))
     
     for event in events:
-        terms[(event.term.description, event.term.id)].append({"name":event.program.programName, "id":event.program.id, "type":"program", "term":event.term})
-    
+        terms[(event.term.description, event.term.id)].append({"name":event.program.programName,
+                                                               "id":event.program.id,
+                                                               "type":"program",
+                                                               "matched": event.reqMatch,
+                                                               "term":event.term.id
+                                                              })
+
     # sorting the terms by the term id
     return dict(sorted(terms.items(), key=lambda x: x[0][1]))
 
+def saveOtherEngagementRequest(engagementRequest):
+    """
+    Create a CommunityEngagementRequest entry based off of the data a user entered for an "Other Engagement Request"
+    """
+    otherEngagementRequest = {"user": engagementRequest['user'],
+                              "experienceName": engagementRequest['experience'],
+                              "term": engagementRequest['term'],
+                              "description": engagementRequest['description'],
+                              "company": engagementRequest['company'],
+                              "weeklyHours": engagementRequest['hours'],
+                              "weeks": engagementRequest['weeks'],
+                              "filename": engagementRequest['attachment'],
```

Comment:
Following up with a previous comment, this should be consistent with whatever is decided in the controller.